Update QueryBuilder.php

sitawit opened the pull request:
Implement resetSequence function using DBCC CHECKIDENT command

sitawit committed:
Update QueryBuilder.php

Implement resetSequence function using DBCC CHECKIDENT command

sitawit commented:
It seems conflict happens because my forked isn't sync with master. Should I resubmit or something?

samdark commented:
No, that's OK.

cebe commented on framework/db/mssql/QueryBuilder.php:
```diff
@@ -71,10 +71,33 @@ public function buildLimit($limit, $offset = 0)
+     * @param array|string $value the value for the primary key of the next new row inserted. If this is not set,
+     * the next new row's primary key will have a value 1.
+     * @return string the SQL statement for resetting sequence
+     * @throws NotSupportedException if this is not supported by the underlying DBMS
```
this line can be removed because this feature now is supported.

cebe commented on framework/db/mssql/QueryBuilder.php:
```diff
@@ -71,10 +71,33 @@ public function buildLimit($limit, $offset = 0)
+     * will have the specified value or 1.
+     * @param string $tableName the name of the table whose primary key sequence will be reset
+     * @param array|string $value the value for the primary key of the next new row inserted. If this is not set,
+     * the next new row's primary key will have a value 1.
```
this comment does not seem to be correct. The implementation finds the maximum + 1 and not just 1.

cebe commented on framework/db/mssql/QueryBuilder.php:
```diff
@@ -71,10 +71,33 @@ public function buildLimit($limit, $offset = 0)
+            $tableName = $this->db->quoteTableName($tableName);
+            if ($value === null) {
+                $key = reset($table->primaryKey);
+                $value = $this->db->createCommand("SELECT MAX(`$key`) FROM $tableName")->queryScalar() + 1;
```
according to the docs at https://msdn.microsoft.com/en-us/library/ms176057.aspx you can even skip this part and just return `DBCC CHECKIDENT ($tableName, RESEED)`, right?

cebe commented:
@sitawit any feedback to my comments?

sitawit commented:
@cebe depends on what you would like it to happen when `value` is null. I copied the same idea from the same function of **MySQL** so it has the same behavior. I'm not sure what is the best practice here so I guess removing it is also alright.

And yes, we can use `DBCC CHECKIDENT ($tableName, RESEED)` solely to change current identity value. 

And you're right about the comment, I didn't read it carefully.

sitawit committed:
Update resetSequence in QueryBuilder.php

Reduce the value of the seed by 1 because the next key be equal to seed + 1.

sitawit committed:
Update resetting sql in QueryBuilder.php of MSSQL

According to this
http://stackoverflow.com/questions/472578/dbcc-checkident-sets-identity-to-0

The current command has a strange feature where it uses a different value in case of newly created table and a used table.

samdark commented:
Need tests to verify it.
